### PR TITLE
spatial/r3: add Divergence

### PR DESCRIPTION
**Pull request by soypat**
Please take a look.

Resolves part of https://github.com/gonum/gonum/issues/1809#issuecomment-1140334118

### CODE REVIEW

File: spatial/r3/vector_test.go [hunk 1]
```diff
@@ -254,6 +256,25 @@ func TestRotate(t *testing.T) {
...
 }
 
+func TestDivergence(t *testing.T) {
+	const (
+		tol = 1e-10
+		h   = 1e-2
+	)
+	step := Vec{X: h, Y: h, Z: h}
+	rnd := rand.New(rand.NewSource(1))
+	for _, Case := range vectorFields {
```
Comment (kortschak) [inline spatial/r3/vector_test.go:266]: s/Case/test/g

File: spatial/r3/vector_test.go [hunk 2]
```diff
@@ -288,3 +309,32 @@ func vecApproxEqual(a, b Vec, tol float64) bool {
 		scalar.EqualWithinAbs(a.Y, b.Y, tol) &&
 		scalar.EqualWithinAbs(a.Z, b.Z, tol)
 }
+
+var vectorFields = []struct {
```
Comment (kortschak) [inline spatial/r3/vector_test.go:313]: Please put this in front of the test function. Are you planning to use this for other tests? If not please s/vectorFields/divergenceTests/.

Comment (soypat) [reply spatial/r3/vector_test.go:313]: I'll move them. I am planning on using them for other tests that require vector fields. i.e. jacobian calculation.

File: spatial/r3/vector.go [hunk 3]
```diff
@@ -83,6 +83,18 @@ func Cos(p, q Vec) float64 {
 	return Dot(p, q) / (Norm(p) * Norm(q))
 }
 
+// Divergence returns the volume density of the outward flux of a vector
```
Comment (vladimir-ch) [inline spatial/r3/vector.go:86]: The current comment unnecessarily focuses on the physical interpretation of the divergence. A user of the function presumably knows what the divergence is. I'd replace it with something like

```
// Divergence returns the divergence of the vector field fn at the point p,
// approximated using central differences with step sizes given in step.
```

Comment (kortschak) [reply spatial/r3/vector.go:86]: With reference to https://github.com/gonum/gonum/pull/1816#discussion_r930844923, s/fn // in the comment and change the parameter name to `field`.

**Review by kortschak — APPROVED** [review]
LGTM but please wait for @vladimir-ch

**Review by vladimir-ch — APPROVED** [review]

**Pull request merged**